README.md

```diff
@@ -119,36 +121,106 @@ export default Total;
+const selectorWithProps = createSelector(
+  state => state.values.value1,
+  state => state.values.value2,
+  (value1, value2, props) => value1 + value2 + props.value3
 );
 
+// A selector created with createSelector ignores the props for memoization
+// See note at bottom of `defaultMemoizeFunc`
+let called = 0;
+const selector = createSelector(
+  state => state.a,
+  (a, props) => {
+    called++;
+    return a + b
+  }
+);
+assert.equal(selector({a: 1}, 100), 101);
+assert.equal(selector({a: 1}, 200), 101);
+assert.equal(called, 1);
+assert.equal(selector({a: 2}, 200), 202);
+assert.equal(called, 2);
 ```
-### createSelectorCreator(valueEqualsFn)
-Return a selectorCreator that creates selectors with a non-default valueEqualsFn. The valueEqualsFn is used to check if the arguments to a selector have changed. The default valueEqualsFn function is:
+
+### defaultMemoizeFunc(func, valueEquals = defaultValueEquals)
+
+`defaultMemoizeFunc` has a cache size of 1. This means it always recalculates when an argument changes, as it only stores the result for immediately preceding value of the argument.
+
+`defaultMemoizeFunc` determines if an argument has changed by calling the valueEquals function. The default `valueEquals` function checks for changes using reference equality:
+
 ```js
-function defaultValueEquals(a, b) {
-  return a === b;
+function defaultValueEquals(currentVal, previousVal) {
+  return currentVal === previousVal;
 }
 ```
+
+```js
+  let called = 0;
+  const memoized = defaultMemoize(state => {
+    called++;
+    return state.a;
+  });
+  const o1 = {a: 1};
+  const o2 = {a: 2};
+  assert.equal(memoized(o1, {}), 1);
+  assert.equal(memoized(o1, {}), 1);
+  assert.equal(called, 1);
+  assert.equal(memoized(o2, {}), 2);
+  assert.equal(called, 2);
+```
+
+`defaultMemoizeFunc` does not look at the props argument when determining if the arguments have changed. This is because by default `defaultMemoize` uses `defaultValueEquals`, and as`defaultValueEquals` is a reference equality check and the reference of the props will be different each render the memoization will not work. 
+
+(NOTE: I am not happy about this, I think this is going to surprise people. I am considering adding an option that specifies whether the props should be used for memoization as part of the configuration options for people who aren't using the reference equality of `defaultValueEquals`. To make things worse, when using a third party memoize like `lodash.memoize` with `createSelectorCreator`, the memoize function **will** take the props into account for memoization. I think it is likely that if you are using a third party memoize your hashing function will be doing some kind of deepEquals so the selector will only update when there is actually a new value somewhere in the props. However, this could still be a problem if the props are changing in a manner such that they resolve to a new never-before-seen hash on every render. Ideas are welcome here.)
```

Comment:
Could the suggestion of line 145 mitigate this problem.  
We could go even further to allow arbitrary number of arguments to the selection function and the selectors determining the arguments for the memoized function have to deal with the selection of the correct properties. Therefore only in case the actual input properties changes does the selected value changes. 
An alternative approach would be to use a compound object of the form ({ props, state }). The latter approach wouldn't force us to change the signature of the select function at all. It would furthermore allow us the to provide convenience selectors like selectWithProp('state', stateSelector), selectWithProp('props', propertySelector).....

I think this approach could enable reselect to achieve the same semantics as if mapStateToProps would be changed into a curried version props => state => selectionFunction

@ellbee, @faassen, @gaearon  What do you think?